mirrord/kube/src/api/kubernetes.rs

```diff
@@ -91,89 +74,93 @@ impl KubernetesAPI {
         }
         Ok(())
     }
+}
 
-    /// Checks if any `ContainerStatus` matches a mesh/sidecar name from our `MESH_LIST`, and the
-    /// user is running incoming traffic in `IncomigMode::Mirror` mode, printing a warning if it
-    /// does.
-    #[tracing::instrument(level = "trace", ret, skip(self, progress))]
-    pub async fn detect_mesh_mirror_mode<P>(
-        &self,
-        progress: &mut P,
-        incoming_mode: IncomingMode,
-        is_mesh: bool,
-    ) -> Result<()>
-    where
-        P: Progress + Send + Sync,
-    {
-        if matches!(incoming_mode, IncomingMode::Mirror) && is_mesh {
-            progress.warning(
-                "mirrord has detected that you might be running on a cluster with a \
-                 service mesh and `network.incoming.mode = \"mirror\"`, which is currently \
-                 unsupported. You can set `network.incoming.mode` to \"steal\" (check out the\
-                 `http_filter` configuration value if you only want to steal some of the traffic).",
-            );
-        }
-        Ok(())
-    }
+#[cfg(not(feature = "incluster"))]
+pub trait UnpinStream:
```

Comment:
:memo: 

Docs on this helper trait.

Also, what is the deal with the `#cfg` for this? Docs for that please.

Reply:
:memo: 

`impl<T> UnpinStream for T where` could you also doc who `T` is, please?

Reply:
Added a doc for the `incluster` feature (crate level)